sourcetracker/_plot.py

```diff
@@ -10,16 +10,271 @@
+        The reason I do not in this case is that these ranges
+        are not particularly helpful to distinguishing the
+        successful matches to each other.
+        vmin=0, vmax=1.0,
+        """
+        prop = self.mpm
+        if not unknowns:
+            prop = prop.drop(['Unknown'], axis=1)
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        if normalized:
+            prop = prop.div(prop.sum(axis=0), axis=1)
+        tra = ""
+        if transpose:
+            prop = prop.T
+            tra = "_Transposed"
+
+        """
+        "viridis" "icefire" "vlag" "Spectral" "mako" "magma"
+        "coolwarm" "rocket" "flare" "crest"
+        "_r" reverses all of these
+        https://seaborn.pydata.org/tutorial/color_palettes.html
+        <-more color palattes here
+        https://matplotlib.org/stable/tutorials/colors/colormaps.html
+        <-and here
+        vmin=0, vmax=1.0,
+        """
+        midpoint = len(prop.columns)/2
+        midpoint = round(midpoint)
+        ratios, g, axes = [], [], []
+        for i in range(len(prop.columns)):
+            ratios.append(1)
+            axes.append("ax" + str(i))
+            g.append("g" + str(i))
+        ratios.append(0.08)
+        axes.append("axcb")
+        fig, axes = plt.subplots(1, len(axes),
+                                 gridspec_kw={'width_ratios': ratios},
+                                 figsize=((prop.shape[1] * 3 / 4)+4,
+                                          (prop.shape[0] * 3 / 4)+4))
+        for i in range(len(prop.columns)):
+            if i == 0:
+                g[i] = sns.heatmap(prop.iloc[:, i:i + 1],
+                                   vmin=0, cmap=self.color,
+                                   cbar=False, annot=annot, ax=axes[i])
+                g[i].set_xlabel("")
+                g[i].set_ylabel(ylabel)
+            elif i == midpoint:
+                g[i] = sns.heatmap(prop.iloc[:, i:i+1], vmin=0,
+                                   cmap=self.color, cbar=False,
+                                   annot=annot, ax=axes[i])
+                g[i].set_xlabel("")
+                g[i].set_ylabel("")
+                g[i].set_yticks([])
+                g[i].set_title(self.title)
+            elif i == len(prop.columns) - 1:
+                g[i] = sns.heatmap(prop.iloc[:, i:i + 1], vmin=0,
+                                   cmap=self.color, annot=annot,
+                                   ax=axes[i],
+                                   cbar_ax=axes[i + 1])
+                g[i].set_xlabel("")
+                g[i].set_ylabel("")
+                g[i].set_yticks([])
+            else:
+                g[i] = sns.heatmap(prop.iloc[:, i:i + 1],
+                                   vmin=0, cmap=self.color,
+                                   cbar=False, annot=annot,
+                                   ax=axes[i])
+                g[i].set_xlabel("")
+                g[i].set_ylabel("")
+                g[i].set_yticks([])
```

Comment:
This seems like a lot of code duplication. I would refactor so that your variables are getting changed in the branches. This would stop you from having to call heatmap multiple times

Reply:
What do you need to address this?